### PR TITLE
Move next-bundle-analyzer to Next.js repo

### PR DESCRIPTION
**Pull request by ijjk**
Moving over `next-bundle-analyzer` from next-plugins

### CODE REVIEW

File: packages/next-bundle-analyzer/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module.exports = (nextConfig = {}) => {
+  return Object.assign({}, nextConfig, {
+    webpack (config, options) {
+      const { isServer } = options
+      const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer')
+
+      if (nextConfig.analyze) {
+        config.plugins.push(
+          new BundleAnalyzerPlugin(
+            Object.assign(
```
Comment (timneutkens) [inline packages/next-bundle-analyzer/index.js]: There's no need for Object.assign here if we don't merge anything else.

Let's use this config:

```
analyzerMode: 'static',
reportFilename: options.isServer ? '../analyze/server.html' : './analyze/client.html'
```

Comment (ijjk) [reply packages/next-bundle-analyzer/index.js]: Updated

File: packages/next-bundle-analyzer/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module.exports = pluginOptions => (nextConfig = {}) => {
```
Comment (Timer) [inline packages/next-bundle-analyzer/index.js]: Shouldn't we default `enabled` to true?

File: packages/next-bundle-analyzer/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  return Object.assign({}, nextConfig, {
+    webpack (config, options) {
+      const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer')
```
Comment (Timer) [inline packages/next-bundle-analyzer/index.js]: Should this be inside the following if statement?

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/next-bundle-analyzer/index.js (hunk 3 above)
Comment (ijjk) [reply packages/next-bundle-analyzer/index.js]: Yeah, good catch

File: packages/next-bundle-analyzer/index.js (hunk 2 above)
Comment (ijjk) [reply packages/next-bundle-analyzer/index.js]: Updated

File: packages/next-bundle-analyzer/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+module.exports = (pluginOptions = { enabled: true }) => (nextConfig = {}) => {
```
Comment (Timer) [inline packages/next-bundle-analyzer/index.js]: I believe this should read out like so:
```suggestion
module.exports = ({ enabled = true } = {}) => (nextConfig = {}) => {
```

This sets `enabled` to true when missing, _and_ supports the null object case.

File: packages/next-bundle-analyzer/index.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  return Object.assign({}, nextConfig, {
+    webpack (config, options) {
+      if (pluginOptions.enabled) {
```
Comment (Timer) [inline packages/next-bundle-analyzer/index.js]: Since we destructure this up top we can just do this now:

```suggestion
      if (enabled) {
```

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]

**Pull request merged**